java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java

```diff
@@ -363,12 +364,15 @@ void process(StorageEventRequest request,
 
     }
 
+    boolean compressedSource(String fileName, String contentEncoding) {
+        return StringUtils.equals(contentEncoding, CONTENT_ENCODING_GZIP) || fileName.endsWith(".gz");
```

Comment:
should match canonical/standards compliant behavior - relying ONLY on content encoding; not using file extension to control the behavior (our code will warn if `.gz` ending w non-gzip encoding)